session/sessionstate.go

```diff
@@ -601,6 +602,20 @@ func (state *State) TriggerContextChanges(ctx context.Context, actionState *acti
 	state.TriggerEvents(actionState, cl.Changed, cl.Closed)
 }
 
+type (
+	TemplateArtifactMap struct {
+		artifactMap *ArtifactMap
+	}
+)
+
+func (artifacts *TemplateArtifactMap) GetIDByTypeAndName(artifactType, name string) (string, error) {
+	artifact, err := artifacts.artifactMap.Lookup(artifactType, name, ArtifactEntryCompareTypeName)
```

Comment:
probably should have a check if artifacts.artifactMap != nil , if it for some reason was nil when added in GetSessionVariable

Reply:
Yeah, I'll add that as a safeguard if artifactmap is not set for any reason.